cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/preview/DefaultPreviewManager.java

```diff
@@ -171,7 +171,9 @@ public ApplicationId start(NamespaceId namespace, AppRequest<?> appRequest) thro
     try {
       runner.startPreview(new PreviewRequest<>(getProgramIdFromRequest(previewApp, appRequest), appRequest));
     } catch (Exception e) {
-      LOG.error("Failed to start the preview run");
+      if (runner instanceof Service) {
+        ((Service) runner).stopAndWait();
```

Comment:
I think its better to catch the `Exception` thrown by this method so that removePreviewDir will always get executed.